### DISCUSSION TITLE
Set different color for cursor when it is at the end of a line?

### DISCUSSION THREAD

**Discussion by michaeljones** [helix-editor/helix#5451] [Ideas]
I'm really enjoying using Helix and I'm grateful for the project. 

I've transitioned from vim and one thing that catches me out is that when the cursor is at the end of the line, I guess, it is selecting the newline and so pressing 'a' goes on to the next line. Or if the line is empty then you're selecting the newline so 'x' selects the line below instead of just the current line. 

I can see why there happen and I don't think it is a good idea to change them to be more contained in the line but I wonder if it might be useful to be able to have the cursor a different color when it is on the newline at the end of a line so that there is some visual indication that some things will behave a little differently that one might unthinkingly expect. 

I'm not a UX designer. I don't know if there might be other ways of approaching this. I'm not confident in this suggestion but I'm just trying to think of things that might allow me to make mistakes less often in this situation.

I do have a new line white space character so I can technically see when the cursor is on the end of the line but it isn't a huge visual clue.

**Comment by michaeljones** [comment]
Made this a while ago but thought of it again today. Variants might be:
- Change the cursor colour and selection colour if the whole of the line that the cursor is on is selected. 
  This would help understand when 'x' is going to select more of the same line or the next line
- Change the cursor colour and selection colour if the current selection includes a new line in any way. Not sure it is as useful but might end up giving the necessary information to the user to avoid the confusion.

**Comment by David-Else** [comment]
I suggest you change your config to make the newline visible (defaults to "⏎"), and when you are used to it, you can remove it again if you like: https://docs.helix-editor.com/configuration.html#editorwhitespace-section

**Reply by michaeljones** [reply]
Thank you for the suggestion. I appreciate it. I do have the following in my config:
```toml
[editor.whitespace.characters]
newline = "¬"
tab = "▸"
```
But for some reason my brain does not process when the selection is at the end of the line and includes the "¬" symbol. I wish it did. I hope that maybe more visual clues will help but perhaps I just try to become more attentive to the details.

**Comment by michaeljones** [comment]
I've just learned about 'shift-x' selecting only the current line and not extending the selection to the next one and I imagine that improves things significantly if I can get used to using that :) 

Probably avoids the need for my original request in the first place.

**Reply by quomat** [reply]
You can interchange S-x and x in the config :)
